tests/phpunit/integration/Core/Util/Migration_1_8_1Test.php

```diff
@@ -0,0 +1,249 @@
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Google\Site_Kit\Core\Authentication\Profile;
+use Google\Site_Kit\Core\Authentication\Verification_File;
+use Google\Site_Kit\Core\Authentication\Verification_Meta;
+use Google\Site_Kit\Core\Authentication\Clients\OAuth_Client;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Util\Migration_1_8_1;
+use Google\Site_Kit\Tests\Fake_Site_Connection_Trait;
+use Google\Site_Kit\Tests\TestCase;
+
+class Migration_1_8_1Test extends TestCase {
+	use Fake_Site_Connection_Trait;
+
+	const FILE_VERIFICATION = Verification_File::OPTION;
+	const META_VERIFICATION = Verification_Meta::OPTION;
+
+	/**
+	 * @var Context
+	 */
+	protected $context;
+
+	/**
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * @var User_Options
+	 */
+	protected $user_options;
+
+	/**
+	 * @var Authentication
+	 */
+	protected $authentication;
+
+	/**
+	 * @var Migration_1_8_1
+	 */
+	protected $migration;
+
+	private $temp_received_identifiers;
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->context        = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options        = new Options( $this->context );
+		$this->user_options   = new User_Options( $this->context );
+		$this->authentication = new Authentication( $this->context, $this->options, $this->user_options );
+		$this->migration      = new Migration_1_8_1( $this->context, $this->options, $this->user_options, $this->authentication );
+
+		$this->temp_received_identifiers = null;
+	}
+
+	public function test_register() {
+		remove_all_actions( 'admin_init' );
+		$this->migration->register();
+
+		$this->assertTrue( has_action( 'admin_init' ) );
+	}
+
+	public function test_migrate_without_using_proxy() {
+		$this->fake_site_connection();
+		$this->migration->migrate();
+		$this->assertNotEquals( Migration_1_8_1::DB_VERSION, $this->get_db_version() );
+	}
+
+	public function test_migrate_without_credentials() {
+		$this->migration->migrate();
+		$this->assertNotEquals( Migration_1_8_1::DB_VERSION, $this->get_db_version() );
+	}
+
+	public function test_migrate_without_users() {
+		$this->fake_proxy_site_connection();
+
+		$this->migration->migrate();
+		$this->assertEquals( Migration_1_8_1::DB_VERSION, $this->get_db_version() );
+	}
+
+	public function test_migrate_without_problem_cases() {
+		$this->create_user( 'administrator', self::FILE_VERIFICATION );
+		$this->create_user( 'subscriber', false );
+
+		$this->fake_proxy_site_connection();
+
+		$this->migration->migrate();
+		$this->assertEquals( Migration_1_8_1::DB_VERSION, $this->get_db_version() );
```

Comment:
Should also test that:
* admin still has Site Kit options.
* no API request was made